detox/test/e2e/17.datePicker.test.js

```diff
@@ -19,9 +19,20 @@ describe(':ios: DatePicker', () => {
       }
     });
 
-    it('can select dates on a UIDatePicker', async () => {
+    it(':ios: can select dates on a UIDatePicker', async () => {
       await element(by.id('datePicker')).setDatePickerDate('2019-02-06T05:10:00-08:00', "ISO8601");
       await expect(element(by.id('utcDateLabel'))).toHaveText('Date (UTC): Feb 6th, 2019');
       await expect(element(by.id('utcTimeLabel'))).toHaveText('Time (UTC): 1:10 PM');
     });
+
+    it(':android: can select dates on a UIDatePicker', async () => {
+      await element(by.id('openAndroidDatePicker')).tap();
+
+      //rn-datepicker does not support testId's on android, so by.type is the only way to match the datepicker right now
+      //@see https://github.com/react-native-datetimepicker/datetimepicker#view-props-optional-ios-only
+      await element(by.type('android.widget.DatePicker')).setDatePickerDate('2019-02-06T05:10:00-08:00', "ISO8601");
```

Comment:
Actually I am wondering... 🤔 What result is supposed to be if I change a timezone to `+08:00` → 5th or 6th February?

Reply:
If I understand it right, for iOS, the devices timezone determines this. So it would make sense to also let the devices timezone decide that on Android, I guess?

My current implementation for android uses the testers locale which is inconsistent.

We could try to use something like this:
```
import java.time.LocalDateTime;
LocalDateTime.parse(date)
```

To parse the date in java which should parse it using the devices timezone.

Reply:
So after some research I think we need to use ZonedDateTime instead. But I am neither fluent in timezone handling nor in android programming.

I just pushed a commit that does what I think might be a solution. But please have a look at it and tell me what you think about it..

Reply:
My concern with this approach would be that we support api 21+, and `ZonedDateTime` was only introduced in api 26. I'm investigating whether we can use `coreLibraryDesugaring`, or whether it would be better to use another alternative

Reply:
Great catch @jonathanmos, if this solution won't work, we can enable this feature only when running with API level 28 or higher. For lower API level, we can throw an exception to indicate that this feature is not available before API 28.

We should have the freedom to increase the minimum API level of our app as long as it is still compatible with the minimum supported version of React Native. However, since React Native currently supports API level 21 and above, we should not increase the minimum API level support in our app beyond that.

Reply:
While desugaring the aar does work, it forces the app to also apply desugaring. 

Let's put an assertion before ZonedDateTime that verifies that the api is 26+ and if not then throw an informative message. This will also require a small change to the documentation for the feature.

```
if (Build.VERSION.SDK_INT < 26) {
            throw new AssertionError("To use DatePicker on Android you must use API 26 or above");
}

Reply:
Indeed got catch. I added your suggestion.

Reply:
@jonathanmos @mauricedoepke 

1. In the current implementation, I see two different parsing strategies: `ZonedDateTime.parse(dateString)` and `LocalDate.parse(dateString, DateTimeFormatter.ofPattern(formatString))`. Are they **both unavailable** for API <26? If only one is unavailable, does it make sense to throw the assertion error in both cases? Maybe we could inform that only `ISO8601` is not supported on older APIs, WDYT? 🤔  
2. @jonathanmos, forgive me for unfamiliarity with the subject, but isn't the suggested runtime assertion `if (Build.VERSION.SDK_INT < 26) {` happening too late? I mean, this `DetoxAction.java` has an import of ZonedDateTime at the very top of the file. Don't we get an error right on an attempt to load that class? 🙏 Thanks in advance for the explanation.

Reply:
I think we could try to use [simpledateformat](https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html) for custom formats instead. This api is much older and should be supported by most devices

Reply:
Looks like simpledateformat can support both custom and iso strings. I rewrote it. That should be supported from api level 1.

Reply:
@jonathanmos I close this discussion now, as I think, using simpledateformat fixes all the here discussed issues. Feel free to open it again if you disagree.